src/poll.c

```diff
@@ -34,6 +34,18 @@ static int luv_new_poll(lua_State* L) {
   return 1;
 }
 
+static int luv_new_socket_poll(lua_State* L) {
+  int fd = luaL_checkinteger(L, 1);
```

Comment:
Well, what I said is wrong, ints are usually 32-bit, but pointers are 64-bit on x86_64 and amd64.  But my comment about using ints for FDs is still true.